packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,196 @@
+        itemToString={itemToString}
+        onChange={onChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          inputValue,
+          isOpen
+        }) => (
+          <div className="ds-u-clearfix ds-c-autocomplete">
+            {this.renderChildren(getInputProps)}
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                {label &&
+                  !loading && (
+                    <h5
+                      className="ds-u-margin--0 ds-u-padding--1 ds-u-padding-left--2"
+                      id={this.labelId}
+                    >
+                      {label}
+                    </h5>
+                  )}
+
+                <ul
+                  aria-labelledby={label ? this.labelId : null}
+                  className="ds-c-list--bare"
+                  role="listbox"
+                >
+                  {this.filterItems(
+                    items,
+                    inputValue,
+                    getInputProps,
+                    getItemProps,
+                    highlightedIndex
+                  )}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
+              aria-label={ariaClearLabel}
+              className="ds-u-float--right ds-u-padding-right--0"
+              onClick={clearSelection}
+              size="small"
+              variation="transparent"
+            >
+              {clearInputText}
+            </Button>
+          </div>
+        )}
+      />
+    );
+  }
+}
+
+Autocomplete.defaultProps = {
+  ariaClearLabel: 'Clear typeahead and search again',
+  clearInputText: 'Clear search'
+};
+
+Autocomplete.propTypes = {
+  /**
+   * Screenreader-specific label for the Clear input link. Intended to provide a longer, more descriptive explanation of the link's behavior.
+   */
+  ariaClearLabel: PropTypes.string,
+  children: PropTypes.node,
+  /**
+   * Clear link text that will appear on the page as part of the rendered component
+   */
+  clearInputText: PropTypes.node,
+  /**
+   * Used to determine the string value for the selected item (which is used to compute the `inputValue`.
+   *
+   * Also see: https://github.com/paypal/downshift#itemtostring
+   */
+  itemToString: PropTypes.func,
+  /**
+   * Array of objects used to populate the suggestion list that appears below the input as users type. This array of objects is intended for an async data callback, and should conform to the prescribed shape to avoid errors.
+   */
+  items: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+      name: PropTypes.string
+    })
+  ).isRequired,
+  /**
+   * Adds a heading to the top of the autocomplete list. This can be used to convey to the user that they're required to select an option from the autocomplete list.
+   */
+  label: PropTypes.node,
+  loading: PropTypes.bool,
```

Comment:
This `loading` prop could benefit from a short description 👼

Reply:
Done. Added descriptions to `loading` and the two new `PropType.node` props for handling Loading and No results language.